src/edu/csus/ecs/pc2/core/model/Problem.java

```diff
@@ -2002,4 +2048,131 @@ public void setVivaInputValidatorHasBeenRun(boolean vivaInputValidatorHasBeenRun
         }
         vivaSettings.setVivaInputValidatorHasBeenRun(vivaInputValidatorHasBeenRun);
     }
+
+    /**
+     * Returns the currently configured memory limit (in MB) for this Problem.
+     * A memory limit of zero indicates "no limit".
+     * Note that memory limits are not enforced unless a sandbox has been selected 
+     * on the Edit Problem GUI (or via YAML configuration).
+     * 
+     * @return
+     */
+    public int getMemoryLimitMB() {
+        return memoryLimitMB;
+    }
+    
+    /**
+     * Sets the memory limit for this problem. Setting a memory limit of zero indicates "no limit",
+     * meaning that the problem is constrained only by the memory provided by the hardware, the OS,
+     * and the specific language runtime system.
+     * 
+     * Note that setting a memory limit does not automatically imply that such limit is enforced; 
+     * enforcing a memory limit requires selection of a problem sandbox capable of doing that.
+     * (See {@link EditProblemSandboxPane}.)
+     * 
+     * If a value less than zero is passed in the memory limit is set to zero (unlimited).
+     * 
+     * @param memLimitInMB the memory limit for the problem, in MB; must be >= 0, where 0=unlimited.
+     */
+    public void setMemoryLimit(int memLimitInMB) {
+        if (memLimitInMB < 0) {
+            this.memoryLimitMB = 0;
+            //TODO: pass a Log into the Problem constructor so conditions like this can be logged properly.
+//            getLog().warning("Memory limit < 0 specified; setting to 0 (unlimited)");
+        } else {
+            this.memoryLimitMB = memLimitInMB;
+        }
+    }
+
+    /**
+     * Returns a String containing the name of the sandbox program associated with this Problem.
+     * Note that the value returned by this method is only relevant if the value returned by
+     * {@link #getSandboxType()} is something other than {@link SandboxType#NONE}.
+     * 
+     * @return the currently-defined sandbox program name.
+     */
+    public String getSandboxProgramName() {
+        if (sandboxProgramName == null) {
+            sandboxProgramName = Constants.PC2_INTERNAL_SANDBOX_PROGRAM_NAME;
```

Comment:
Hmmm... I wonder about the above two lines (and yes, I'm aware that it's likely that I wrote them... over a year ago).  It doesn't seem quite right that if no sandbox program name has been defined then the name of the sandbox should be changed.  Further, this is an example of a getter with a side-effect, which isn't good style (and again, yes, I'm aware I probably wrote it -- here, I'm proving that I'm willing to acknowledge my errors :) ).  I'm thinking that if the current sandbox program name is null, that's probably what should be returned -- unless someone can come up with a good argument why calling a getter should cause a _change_ in the problem configuration.

Reply:
I am reluctant to make a major change like this at this point in time.  There are a few other classes that use these methods and **expect** them to return "something" other than `null`.  (`ProblemsReport`, `EditProblemSandboxPane`, ExportYAML, `Executable`).    Returning the internal sandbox program (or sandbox command line)  if it is not set is not right either.  It would seem to be more correct to return an empty string, "", instead of `null` - this I can do and it should have minimal impact.  The problem is returning `null`.   Does anyone have any opinions?   For the record, in our normal use case (when using a sandbox), the program name and command line are set when the sandbox type is set (setting the type to `PC2_INTERNAL_SANDBOX_COMMAND_LINE `sets those members).  When using a CDP to load a contest, If the sandbox type is `NONE `or the currently unsupported `EXTERNAL_SANDBOX`, then those members are initialized to empty strings (not null), and, if values are supplied from the CDP, they are filled in.  They are only `null `if a contest is not loaded from yaml files (that is, hand configured), in which case they appear as the `INTERNAL `values currently since the getter methods will return those instead of `null`.

Reply:
I think that returning an empty string (instead of null) is probably the best approach at this point.  I do agree that doing anything more could have unanticipated consequences...